### ISSUE TITLE
java.lang.IllegalStateException: Already building!

### ISSUE THREAD

**Issue by iarspider** [Lunatrius/Laser-Level#4]
Game crashes when joining a server.

Crash report: http://pastebin.com/jdTByUny
Forge 1721, LunatriusCore 1.1.2.32, Laser Level 2.0.0.8

**Comment by iarspider** [comment]
As a side note: Laser Level mod is missing from this page - http://mc.lunatri.us/wiki/misc:crashreports

**Comment by Lunatrius** [comment]
Does it happen every time or was it a one time crash? Did you go straight into MP or were you previously in SP or anywhere else?

**Comment by iarspider** [comment]
Every time. Straight to MP.

**Comment by Lunatrius** [comment]
Could you test SP to check if it's reproducible in SP? I'm not entirely sure why this can happen - Laser Level should be the only mod accessing that particular `WorldRenderer` instance that is causing the crash.

Is that a public mod pack that I could use to test this myself?

**Comment by iarspider** [comment]
Tested in SP, crashes as well. You can grab my modpack (MultiMC instance) here: https://www.dropbox.com/s/1i23yuz3jx11xi0/NotFC.zip?dl=0

**Comment by Lunatrius** [comment]
I'm unable to reproduce the crash - I tried different settings that could affect it and a super flat and normal world. My suggestion would be to clone the instance and disable all of the following mods:

```
WARNING: coremods are present:
  ThaumcraftLoader (Thaumcraft-1.8.9-5.1.4.jar)
  DCLoadingPlugin (DummyCore-2.1.189.12.jar)
  FldLoadingPlugin (FastLeafDecay-1.8-1.6.jar)
  LoadingPlugin (RandomThings-MC1.8.9-3.5.8.jar)
  CorePlugin (Fluxed Redstone-1.0.0.3.jar)
  thaumicinfusion (ThaumicInfusion-1.8.9-4.76.jar)
  BIFLoader (BowInfinityFix-1.8.x-rv2.jar)
  ShadowMCCore (ShadowMC-1.8.9-3.0.0-SNAPSHOT.jar)
  TrashSlot (trashslot-mc1.8.9-2.0.7.jar)
Contact their authors BEFORE contacting forge
```

If it works, enable them one by one to figure out which core mod is incompatible.

If it doesn't, try disabling all mods except LunatriusCore and Laser Level and test it. If that works, keep enabling a few mods at a time (write down the names) and repeat until you start getting the crash again.

_Alternatively you can disable half of the mods, test, if it works repeat, if it doesn't work invert the selection and repeat - basically a form of [binary search](https://en.wikipedia.org/wiki/Binary_search_algorithm#Example)._

**Comment by iarspider** [comment]
Disabled all the mods (except LunatriusCore and LaserLevel), still crashing. 
Full log: https://paste.ee/p/HwfNH

**Comment by Lunatrius** [comment]
~~That's rather annoying... it really shouldn't be happening. Could you try dropping the two mods and Forge into a brand new profile to check whether it could be a setting that causes this? Also try messing with the VBO setting, it's the only one I can think of right now.~~

I just managed to reproduce it and I have no idea how... turn off VBOs for the time being. Whenever I turn VBOs on it crashes.

**Comment by Lunatrius** [comment]
Fixed in 6e7432d40aa16dd86f22178bb54fc900fa582c88. I'm not ever sure how it ever worked as it was...
